add_route.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+#
+# add_route.sh
+# Script to setup persistant TEDAPI Routing for Powerwall Dashboard
+# Version: 1.0.0
+# By Scott Hibbard  - 2024-09-14
+#
+
+CRONTAB="/var/spool/cron/crontabs/root"
+PW_IP=""
+SCRIPT_NAME="TDAPI_routing"
+DIR="/root/scripts"
+
+echo "Setup script for persistant Powerall Dashboard TEDAPI Interface network routing"
+echo "-------------------------------------------------------------------------------"
+echo
+echo "This script will require root privileges to read & set a startup cron task."
+read -r -p "Do you want to run this script? [Y/n] " response
+if [[ "$response" =~ ^([nN][oO]|[nN])$ ]]; then
+	echo "Cancel"
+	exit 1
+fi
+
+OS=$(uname -s)
+
+if $(ip route | grep -qw "192.168.91.0/24"); then
+	read -r -p "192.168.91.0/24 routing is already in routing table. Still want to run this? [y/N] " response
+	if [[ ! "$response" =~ ^([yY][eE][sS]|[yY])$ ]]; then
+		echo "Cancel"
+		exit 1
+	fi
+fi
+
+# Running Linux?
+if [[ "${OS}" == "Linux" ]]; then
+	if ! sudo test -d ${DIR}; then
+		sudo mkdir ${DIR}
+		echo "Created folder ${DIR}."
+	fi
```

Comment:
Oops, ignore my note below. You already had this logic to create the dir! We can replace this "if block" with a single line or leave it to let the user know we created a dir. I would add the `-p` flag just in case the installation is missing /root:

```bash
sudo mkdir -p "${DIR}"
```

Reply:
You were busy yesterday!  Got a few things to do this morning, will take care of the issues later in the day.  Forgot about the -p argument!  Don't think we need to inform if a folder is made, will just use -p.